addons/web/static/tests/views/kanban_tests.js

```diff
@@ -839,6 +839,73 @@ QUnit.module('Views', {
         kanban.destroy();
     });
 
+    QUnit.test('Do not open record when clicking on `a` with `href`', function (assert) {
+        assert.expect(6);
+
+        delete this.data.partner.fields.state;
+        this.data.partner.records = [
+            { id: 1, foo: 'yop' },
+        ];
+
+        var isRecordOpen = false;
+
+        var kanban = createView({
+            View: KanbanView,
+            model: 'partner',
+            data: this.data,
+            arch: '<kanban class="o_kanban_test">' +
+                        '<templates>' +
+                            '<t t-name="kanban-box">' +
+                                '<div class="oe_kanban_global_click">' +
+                                    '<field name="foo"/>' +
+                                    '<div>' +
+                                        '<a class="o_test_link" href="#">test link</a>' +
+                                    '</div>' +
+                                '</div>' +
+                            '</t>' +
+                        '</templates>' +
+                    '</kanban>',
+            intercepts: {
+                // when clicking on a record in kanban view,
+                // it switches to form view.
+                switch_view: function () {
+                    isRecordOpen = true;
+                },
+            },
+        });
+
+        var $record = kanban.$('.o_kanban_record:not(.o_kanban_ghost)');
+        assert.strictEqual($record.length, 1,
+            "should display a kanban record");
+
+        var $testLink = $record.find('a');
+        assert.strictEqual($record.find('a').length, 1,
+            "should contain a link in the kanban record");
+        assert.ok(!!$testLink[0].href,
+            "link inside kanban record should have non-empty href");
+
+        // Mocked views prevent accessing a link with href. This is intented
+        // most of the time, but not in this test which specifically needs to
+        // let the browser access a link with href.
+        kanban.$el.off('click', 'a');
+        // Prevent the browser default behaviour when clicking on anything.
+        // This includes clicking on a `<a>` with `href`, so that it does not
+        // change the URL in the address bar.
+        $(document.body).on('click', 'a', function (ev) {
```

Comment:
you should add a namespaced event handler, and remove it at the end of your test